### PR TITLE
feat: Vote Ticker for Header and Footer (desktop, stub data)

### PR DESCRIPTION
**Pull request by Tulun**
1. Vote Ticker widget
Screen Shot 2022-10-20 at 3 56 10 PM
Screen Shot 2022-10-20 at 3 56 14 PM

2. Some Footer styles / fixes.

### CODE REVIEW

File: components/VoteTicker/VoteTicker.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  const { timeRemaining } = useVoteTicker();
+  return (
+    <Section theme={theme}>
+      <Wrapper theme={theme}>
+        <VoteBlock theme={theme}>
+          <ClockBG theme={theme}>
+            <Clock />
+          </ClockBG>
+          <VoteText theme={theme}>
+            Time to commit vote:
```
Comment (daywiss) [inline components/VoteTicker/VoteTicker.tsx:23]: this probably should be a  prop

File: components/VoteTicker/calculateTimeRemaining.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { DateTime, Duration } from "luxon";
+
+// For Timer feature.
+// Countdown from UTC 00:00:00.
+export default function calculateTimeRemaining() {
```
Comment (daywiss) [inline components/VoteTicker/calculateTimeRemaining.ts:5]: this should be named something like, timeToEndOfDay, and imo id separate the time in ms and the formatting of that time for display into separate functions

**Review by daywiss — APPROVED** [review]
couple nits, not blockers, up to you

File: components/VoteTicker/calculateTimeRemaining.ts (hunk 2 above)
Comment (Tulun) [reply components/VoteTicker/calculateTimeRemaining.ts:5]: I think this function may just be poorly named. In reality it really is just formatting the "UTC time" as we never actually need a time input due to votes always starting at UTC midnight. General case for this is pointless as we should never need a time in MS input.

WDYT?

File: .storybook/main.ts [hunk 3]
```diff
@@ -28,31 +28,33 @@ const config: StorybookConfig = {
...
-          options: {
-            data: `@import "./assets/scss/app.scss";`,
-            includePaths: [__dirname, "./assets/**/*"],
+    config?.module?.rules?.push(
+      {
+        test: /\.svg$/,
+        use: ["@svgr/webpack"],
+      },
+      {
+        test: /\.scss$/,
```
Comment (ryanwolhuter) [inline .storybook/main.ts:37]: what's up with the scss? Ideally we don't want to have both scss and styled components at the same time.

File: components/VoteTicker/VoteTicker.tsx (hunk 1 above)
Comment (Tulun) [reply components/VoteTicker/VoteTicker.tsx:23]: Yeah fair. I guess we need two props here:

1) Enum for commit reveal (similar to return from Contract)
2) Num Votes (determines if this even renders).

WDYT?

File: components/VoteTicker/VoteTicker.tsx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+import UpRightArrow from "public/assets/up-right-arrow.svg";
+import calculateTimeRemaining from "./calculateTimeRemaining";
+
+type theme = "light" | "dark";
+
+interface Props {
+  theme: theme;
+}
+
+const VoteTicker: React.FC<Props> = ({ theme }) => {
```
Comment (ryanwolhuter) [inline components/VoteTicker/VoteTicker.tsx:13]: Using [styled components native theming](https://styled-components.com/docs/advanced) will make your life much easier with this

File: .storybook/main.ts (hunk 3 above)
Comment (Tulun) [reply .storybook/main.ts:37]: I have a style override for Sandpack.

File: components/VoteTicker/VoteTicker.tsx [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+function useVoteTicker() {
+  const [timeRemaining, setTimeRemaining] = useState("00:00");
+  // Set time remaining depending if it's the Commit or Reveal
+  // Note: the requests are all slightly differently in there final vote time. I'll use the last
+  // Vote added.
+  useEffect(() => {
+    setTimeRemaining(calculateTimeRemaining());
+
+    const timer = setInterval(() => {
```
Comment (ryanwolhuter) [inline components/VoteTicker/VoteTicker.tsx:48]: Doing this in a hook is going to create new intervals every time the hook gets called in a component, which might not be what you want. [This](https://usehooks-ts.com/react-hook/use-interval) could help.

File: components/VoteTicker/VoteTicker.tsx [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+  return { timeRemaining };
+}
+
+interface IStyledProps {
+  theme: theme;
+}
+
+const Section = styled.div<IStyledProps>`
+  width: 100%;
+  background: ${({ theme }) => (theme === "dark" ? "inherit" : "var(--grey-100)")};
```
Comment (ryanwolhuter) [inline components/VoteTicker/VoteTicker.tsx:63]: Also the styled components built in theming will be much better here

**Review by ryanwolhuter — APPROVED** [review]
lgtm

File: components/VoteTicker/VoteTicker.tsx (hunk 1 above)
Comment (daywiss) [reply components/VoteTicker/VoteTicker.tsx:23]: id put the condition of rendering into the parent, and probably just pass a string prop for the label, it doesnt need to be strongly typed imo, but if u want to make an enum thats fine

File: components/VoteTicker/calculateTimeRemaining.ts (hunk 2 above)
Comment (Tulun) [reply components/VoteTicker/calculateTimeRemaining.ts]: Renamed to be much clearer.

File: components/VoteTicker/VoteTicker.tsx (hunk 4 above)
Comment (Tulun) [reply components/VoteTicker/VoteTicker.tsx:13]: I'll dig into this for a potential refactor :)

File: components/VoteTicker/VoteTicker.tsx (hunk 5 above)
Comment (Tulun) [reply components/VoteTicker/VoteTicker.tsx:48]: Love me some good generic hooks, I'll implement this.

**Pull request merged**